code/espurna/config/validate.h

```diff
@@ -0,0 +1,8 @@
+#pragma once
+
+static_assert((strlen(HOSTNAME) == 0) || ((strlen(HOSTNAME) > 1) && (strlen(HOSTNAME) < 31)), "HOSTNAME must be 1..31 characters");
```

Comment:
Sure it can. In case of DNS / SSID specifications use 'up to ...'. Some discussions:
https://serverfault.com/questions/162038/are-one-letter-host-names-valid <- hostnames
https://serverfault.com/a/45509 <- SSID

https://tools.ietf.org/html/rfc1123 <- rfc that initial issue about underscores referenced
https://tools.ietf.org/html/rfc1034#section-3.1 <- dns labels size, same '0 to 63'

Reply:
ok i was looking at the different condition. fixed